refactor: comprehensive code style cleanup/CI fix

modcrafter77 opened the pull request:
- Move Problem types from modkit::http to modkit::api module for better organization
- Apply consistent code formatting and style across entire codebase
- Fix test environment setup for home directory resolution
- Remove unused dependencies and clean up import statements
- Improve error handling patterns and remove redundant code. anyhow is replaced with thiserror for DB library. (clippy proposal)
- Standardize formatting with Unix newlines and consistent indentation
- Update dependencies

claudedigon commented on libs/executor/executor_config.go:
```diff
@@ -0,0 +1,151 @@
+		return err
+	}
+
+	/*
```
will be fixed later in separate patches

modcrafter77 committed:
refactor: comprehensive code style cleanup and module reorganization
- Move Problem types from modkit::http to modkit::api module for better organization
- Apply consistent code formatting and style across entire codebase
- fix clippy errors
- fix licenses issues (deny check)
- Fix test environment setup for home directory resolution
- Remove unused dependencies and clean up import statements
- Improve error handling patterns and remove redundant code. anyhow is replaced with thiserror for DB library.
- Standardize formatting with Unix newlines and consistent indentation
- Update dependencies
- .editorconfig is added to prevent poisoning Linux line-endings
